### PR TITLE
Be resilient to not being able to get a SymbolTreeInfo for a given PEReference

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems?id=274179&_a=edit

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs [hunk 1]
```diff
@@ -453,8 +453,12 @@ internal static class DependentTypeFinder
             // implement that type.  Because the mapping is from the simple name
             // we might get false positives.  But that's fine as we still use 
             // 'metadataTypeMatches' to make sure the match is correct.
-            var symbolTreeInfo = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
+            var symbolTreeInfo = await SymbolTreeInfo.TryGetInfoForMetadataReferenceAsync(
                 project.Solution, reference, loadOnly: false, cancellationToken: cancellationToken).ConfigureAwait(false);
+            if (symbolTreeInfo == null)
+            {
+                return;
+            }
```
Comment (jmarolf) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs]: Can we add a test case for this?

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs]: Not simply.  But if you have a bogus metadata dll, we could attempt to check it in, then run this scenario.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest  please

**Pull request merged**